tingbot/graphics.py

```diff
@@ -217,6 +217,51 @@ def text(self, string, xy=None, color='grey', align='center', font=None, font_si
 
         self.image(text_image, xy=xy, align=align, scale=1)
 
+    def oval(self, xy=None, size=(100,100), color='grey', align='center'):
+        """
+        Draws an oval.
+        
+        Args:
+            xy (tuple): The position (x, y) to draw the oval, as measured from the top-left.
+            size (tuple): The size (width, height) of the oval.
+            color (tuple or str): The color (r, g, b) or color name.
+            align (str): How to align the outside box of the oval relative to `xy`, or relative to the drawing surface
+                if `xy` is None. Defaults to 'center'.
+        """
+        if len(size) != 2:
+            raise ValueError('size should be a 2-tuple')
+
+        xy = _topleft_from_aligned_xy(xy, align, size, self.size)
+
+        pygame.draw.ellipse(self.surface, _color(color), pygame.Rect(xy, size), 0) 
+
+    def circle(self, xy=None, size=100, color='grey', align='center'):
+        """
+        Draws a circle.
+
+        Args:
+            xy (tuple): The position (x, y) to draw the circle, as measured from the top-left.
+            size (int): The diameter of the circle.
+            color (tuple or str): The color (r, g, b) or color name.
+            align (str): How to align the outside box of the circle relative to `xy`, or relative to the drawing surface
+                if `xy` is None. Defaults to 'center'.
+        """
+        if xy is None:
+            raise ValueError('no value given for xy')
+        elif type(xy) is not tuple:
+            raise ValueError('xy should be a 2-tuple')
+        elif len(xy) != 2:
+            raise ValueError('xy should be a 2-tuple')
```

Comment:
Thanks @furbrain, I've had a think and I'd agree. Really the purpose of this type checking is to present more helpful error messages to the user, so we shouldn't raise an exception on anything that would otherwise work.

It's more idiomatic to just let Python raise an exception, but when beginners are presented with an error like `TypeError: 'int' object has no attribute '__getitem__'` there's nothing there that's going to help them debug the problem (and googling it isn't going to help)

How about this for the type check:

```
if not (hasattr(xy, '__len__')
        and len(xy) >= 2
        and isinstance(xy[0], numbers.Real)
        and isinstance(xy[1], numbers.Real)):
    raise TypeError('xy should contain two numbers')
```

---

side note:  
I'm not a big fan of 'type' checking overall, since as a library author you're asserting yourself over the user (assuming they don't know what they're doing). However we need to improve the error messages in tingbot-python to improve accessibility. I think in Python it's possible to do this without relying on `isinstance` (except with numbers, numbers are numbers!).